servicetalk-concurrent-internal/src/main/java/io/servicetalk/concurrent/internal/DelayedCancellable.java

```diff
@@ -52,4 +52,11 @@ public void cancel() {
             oldCancellable.cancel();
         }
     }
+
+    /**
+     * Ignores any subsequent calls to {@link #cancel()}, preventing propagating the cancellation further up the stream.
+     */
+    protected void ignoreCancel() {
```

Comment:
This is going to behave odd wrt ordering of `delayedCancellable` calls. In my minds eye, calling `ignoreCancel()` before or after `delayedCancellable` should have the same behavior, namely that cancellation will be ignored whereas right now if you call `ignoreCancel()` first it's going to result in immediate cancellation of the cancellable passed in through `delayedCancellable(..)`.

Reply:
I would say that the current behavior is correct. We can only ignore if we know that this `Subscription` is already terminated. And in this case it's all correct, because terminal events can come only after `onSubscribe(Cancellable)`. We alway ignore after terminal.
If in some other use-case in the future, there is a path that calls `ignoreCancel()` before `delayedCancellable`, we should immediately cancel any new `Cancellable` because it may hold some resource that should be cleaned up.

Reply:
At a minimum we need to clarify the behavior in the API docs because it reads as if it will do what I was describing.